compute_tools/src/compute.rs

```diff
@@ -1141,8 +1171,26 @@ impl ComputeNode {
     /// Fetches a basebackup via libpq. The connstring must use postgresql://. Returns the timestamp
     /// when the connection was established, and the (compressed) size of the basebackup.
     fn try_get_basebackup_libpq(&self, spec: &ParsedSpec, lsn: Lsn) -> Result<(Instant, usize)> {
-        let shard0_connstr = spec.pageserver_connstr.split(',').next().unwrap();
-        let mut config = postgres::Config::from_str(shard0_connstr)?;
+        let shard0_index = ShardIndex {
+            shard_number: ShardNumber(0),
+            shard_count: spec.pageserver_conninfo.shard_count,
```

Comment:
nit: in the future, it could be a method on the `pageserver_conninfo`, so that we don't manually create this all the time but just invoke a named method without arguments.

Reply:
👍 Same for getting the shard 0 connstring.